### DISCUSSION TITLE
PnP Modern Search with Microsoft Viva Connections - Search Top Bar not available

### DISCUSSION THREAD

**Discussion by ferrarirosso** [microsoft-search/pnp-modern-search#2484] [General]
We have set up our intranet as a home site and are planning to use the MS Viva Connections App within MS Teams (browser, desktop).

We have noticed that the search top bar of our intranet is not rendered when we use the MS Viva Connections app within MS Teams.

This means that our custom intranet result page, where we have all our pnp modern search customizations, is not available when using MS Viva Connections and all the efforts we have put in search customizations are not available with MS Teams...

The MS Teams search bar is available, but it's the search within MS Teams and not within our intranet.

Is this :
1. A bug ?
2. What are the possible solution to keep the pnp modern search customization within the MS Viva Connection App ?

Thanks !

**Comment by wobba** [comment]
1. Not a bug
2. Currently you need a PnP search box on the page

However, we are looking into ways for how search and Viva Connections can work better together.

One question, any way you can use the OOB search experience instead of re-creating it with the PnP web parts? (which is not a solution I would go for due to search on office.com and more cannot be controlled like you do in SharePoint).

**Comment by ferrarirosso** [comment]
Thank you for your quick answer and for looking at a better integration between Search & Viva Connection.

We use the following OOTB configuration :
- Search results redirection to custom page / site which contains all our customizations (pnp search solution / search result sources...)
We do not use the PnP Search Box on our page, apart from our custom search center.

We do not use the OOTB search experience due to the lack of customizations, like you did mention. Same for office.com

At the moment I see the following workarounds
### PnP Search Box in Teams-Mode 
I think we should be able to inject a PnP Search Box or a Search-Box thru a SPFx App Customizer to restore the missing search box functionality

### Search Command Message Extensions in Teams
This is a bit a more advanced scenario. We are looking at a way to implement a search message extension which replicates our searches. I am not actually sure this is technically possible (SPO auth ?) and would certainly require to duplicate a lot of logic from our custom search center.

If you see any other possiblities, let me know

Thanks !

**Reply by wobba** [reply]
I've seen an internal demo using ME and search, so certainly doable. It's all about doing auth and then calling SP/Graph search API's with the token.

**Comment by ferrarirosso** [comment]
@wobba quick follow up, thanks for the implementation, works well, we now have the MS-Teams top search bar which automatically redirect to our search result page !

**Reply by wobba** [reply]
@ferrarirosso you mean as part of Viva Connections in Teams?

**Reply by ferrarirosso** [reply]
@wobba yes

**Reply by wobba** [reply]
I know it went live just now 🙂

**Comment by kasperbolarsen** [comment]
@ferrarirosso if possible, could you blog about this as it certainly is both cutting edge and could be of interest to many people 🙏

**Reply by ferrarirosso** [reply]
@kasperbolarsen on holiday currently, but will consider it and ping you back. Thanks for your answer @kasperbolarsen

**Reply by wobba** [reply]
Maybe I'll drop a blog post on this as well, and have been involved with the work a little bit.

**Reply by ferrarirosso** [reply]
@wobba if you could also elaborate if the search top bar / search results page is also dependent upon the viva connection (there could be more than one), that would be great, thx

**Reply by wobba** [reply]
Search from VivaConnections should show the same results/experience as configured on the Home Site. That said, the top bar in Teams in context of Viva Connection does not reflect the same suggestions as when in SharePoint - it follows what Teams shows. Does this help? Or if can be more concrete I can try to elaborate.

**Reply by ferrarirosso** [reply]
@wobba thanks
My question was not related to the query suggestions but in the top search bar, but there is the possibilites to create up to ten "viva connections".
[mulitple viva connections](https://learn.microsoft.com/en-us/viva/connections/set-up-admin-center)
In such scenarios, does the top search bar "adapt" to the corresponding scenario / viva connection and search in the corresponding search result page ?

**Reply by wobba** [reply]
@ferrarirosso you have to ask the Viva Connection people about their setup and how it works. As a page has multiple top bars you can include a screenshot to illustrate what you mean. But the top suite bar navigation  SharePoint is always hidden when framed inside of Teams - which makes sense. The nav bars below do show, like a HUB navigation.

I have not tried Viva Connection with a custom result page myself - to see if this is possible or not. Another point to why you should not replace the full search experience using PnP modern search - only use it for specific scenarios (my personal opinion).
